### PR TITLE
fix: Fixing METAL NIMF docs

### PR DESCRIPTION
**Pull request by srushti-patl**
Fixing Metal-NIMF doc examples
Reference: https://github.com/equinix/terraform-provider-equinix/issues/744

### CODE REVIEW

File: examples/resources/metal_connection/shared_nimf_to_csp.tf
Comment (ocobles) [inline examples/resources/metal_connection/shared_nimf_to_csp.tf:15]: This condition causes the example to fail when converting to Pulumi and does not provide any value here as this example is just for connecting to a service provider. It would be helpful if you removed the `count` condition here. This update will also require updating the line referencing it `uuid = data.equinix_fabric_service_profiles.zside[0].id` with `uuid = data.equinix_fabric_service_profiles.zside.id`.

File: templates/resources/metal_connection.md.tmpl
Comment (ctreatma) [inline templates/resources/metal_connection.md.tmpl:35]: The example file reference here, as well as the one below, should be renamed to remove `_nimf_`

**Comment by ctreatma** [comment]
LGTM but I'm going to leave it to @ocobles to submit an official review for the Metal parts of this PR since he has a better idea of what needs to be done here.

**Review by ocobles — APPROVED** [review]

**Comment by ctreatma** [comment]
@equinix/governor-digin-fabric this has been approved from a DRE perspective; just requires review and approval from your team before it can be merged.

**Review by thogarty — APPROVED** [review]
LGTM.

**Pull request merged**
